### PR TITLE
Clean before deploying

### LINKED ISSUE #260: ClassNotFoundException when using com.google.gcloud.AuthCredentials in 0.0.9
I've just upgraded to 0.0.9 from 0.0.8-SNAPSHOT and I get
```
java.lang.ClassNotFoundException: net.sourceforge.cobertura.coveragedata.TouchCollector
```
calling
```java
AuthCredentials.createFor()
```

### PR DESCRIPTION
**Pull request by ajkannan**
Fixes #260 

Note: I changed the settings location to ~/.m2/settings.xml for deployment.  I don't spot any difference because this is copied in before_install, but it'd be good to get a second opinion on that.

### CODE REVIEW

**Comment by ajkannan** [comment]
Working on verifying the changes locally, will update soon.

**Comment by mziccard** [comment]
As I said in #262 I am not sure that it is worth removing `-DskipTests=true`. This will cause build to take longer and I am not sure the tests will detect the error we experienced in #260.

Using `~/.m2/settings` *should* do no harm.

**Comment by ajkannan** [comment]
Testing locally, the tests do in fact detect the error experienced in #260, so I'm in favor of keeping it.  It's a good safeguard for the future and only costs ~3-5 minutes in build time.

File: utilities/after_success.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ if [ "${TRAVIS_JDK_VERSION}" == "oraclejdk7" -a "${TRAVIS_BRANCH}" == "master" -
 
         cd ..
         utilities/update_docs_version.sh # Update version in READMEs
-        mvn deploy -DskipTests=true --settings target/travis/settings.xml -P sign-deploy
+        mvn clean deploy --settings ~/.m2/settings.xml -P sign-deploy
     else
         mvn deploy -DskipTests=true -Dgpg.skip=true --settings target/travis/settings.xml
```
Comment (aozarov) [inline utilities/after_success.sh]: lets do the clean (and the tests here as well).

File: utilities/verify.sh [hunk 2]
```diff
@@ -6,12 +6,12 @@
 if [ "${TRAVIS_BRANCH}" == "master" -a "${TRAVIS_PULL_REQUEST}" == "false" ]; then
     # Get signing tools and API keyfile
     openssl aes-256-cbc -K $encrypted_631490ecae8f_key -iv $encrypted_631490ecae8f_iv -in target/travis/signing-tools.tar.enc -out target/travis/signing-tools.tar -d
-    mkdir target/travis/signing-tools
-    chmod 700 target/travis/signing-tools
-    tar xvf target/travis/signing-tools.tar -C target/travis/signing-tools
+    mkdir $TRAVIS_BUILD_DIR/signing-tools
+    chmod 700 $TRAVIS_BUILD_DIR/signing-tools
+    tar xvf target/travis/signing-tools.tar -C $TRAVIS_BUILD_DIR/signing-tools
```
Comment (aozarov) [inline utilities/verify.sh]: s/target/travis/$TRAVIS_BUILD_DIR/ ?

File: utilities/after_success.sh (hunk 1 above)
Comment (mziccard) [inline utilities/after_success.sh]: Are we sure we want to re-compile and re-run the tests?
It seems to me that the cause of the issue is `mvn site` and we do not run it for snapshots. I would avoid making travis build even slower if not strictly necessary.

Comment (aozarov) [inline utilities/after_success.sh]: Ok, I agree that there is no need to re-run the tests for snapshots, though I think we should probably move `mvn cobertura:cobertura coveralls:report` to be done only in the non-snapshots branch (as it is only needed for the site).

Comment (mziccard) [inline utilities/after_success.sh]: Agreed!

Comment (aozarov) [inline utilities/after_success.sh]: actually, we probably want to leave it outside, so we can see the change in Corvallis after every commit. Regardless, fine with no re-run tests for snapshots (fixing a bad snapshot is simple).

**Pull request merged**